File: java/src/main/java/com/techelevator/services/CommentService.java
package com.techelevator.services;


import com.techelevator.dao.CommentDao;
import com.techelevator.model.Comment;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class CommentService {

    private final CommentDao commentDao;

    public CommentService(CommentDao commentDao) {
        this.commentDao = commentDao;
    }

    public List<Comment> getCommentsByReviewId(int reviewId) {
        return commentDao.getCommentsByReviewId(reviewId);
    }

    public Comment getCommentById(int id) {
        return commentDao.getCommentById(id);
    }

    public Comment createComment(Comment comment) {
        return commentDao.createComment(comment);
    }

    public Comment updateComment(Comment comment) {
        return commentDao.updateComment(comment);
    }

    public boolean deleteCommentById(int id) {
        return commentDao.deleteCommentById(id);
    }

}

File: vue/src/services/CommentService.js
import axios from 'axios';

export default {

  getCommentsByReviewId(reviewId) {
    return axios.get(`/review/${reviewId}/comments`);
  },

  getCommentById(id){
    return axios.get(`/comment/${id}`);
  },

  addComment(comment){
    return axios.post(`/comment`, comment)
  },

  editComment(id, comment){
    return axios.put(`/comment/${id}`, comment)
  },

  deleteCommentById(id){
    return axios.delete(`/comment/${id}`)
  }
}